File: jupyter2/ps/variables/point_line_distance/jl/point_line_distance.jl
### begin my answer

### begin hidden
function point_line_distance(x0, y0, x1, y1, p, q)
    dx, dy = x1 - x0, y1 - y0
    a, b = dy, -dx
    num = abs(a * (p - x0) + b * (q - y0))
    den = sqrt(a * a + b * b)
    num / den
end
### end hidden
### end my answer

function main()
    @assert abs(point_line_distance(-3.0, 1.0, 1.0, -2.0, 4.0, -3.0) - 1.0) < 1.0e-5
    @assert abs(point_line_distance( 2.0, 2.0, 4.0,  0.0, 1.0,  0.0) - 2.12132) < 1.0e-5
    @assert abs(point_line_distance( 1.0, 1.0, 4.0,  3.0, 3.0, -2.0) - 3.60555) < 1.0e-5
    println("OK")
end

main()

File: jupyter2/ps/variables/point_line_distance/rs/point_line_distance.rs
/** begin my answer */

/** begin hidden */
fn point_line_distance(x0: f64, y0: f64, x1: f64, y1: f64, p: f64, q: f64) -> f64 {
    let dx = x1 - x0;
    let dy = y1 - y0;
    let a = dy;
    let b = -dx;
    let num = (a * (p - x0) + b * (q - y0)).abs();
    let den = (a * a + b * b).sqrt();
    num / den
}
/** end hidden */
/** end my answer */

fn main() {
    assert!((point_line_distance(-3.0, 1.0, 1.0, -2.0, 4.0, -3.0) - 1.0).abs() < 1.0e-5);
    assert!((point_line_distance( 2.0, 2.0, 4.0,  0.0, 1.0,  0.0) - 2.12132).abs() < 1.0e-5);
    assert!((point_line_distance( 1.0, 1.0, 4.0,  3.0, 3.0, -2.0) - 3.60555).abs() < 1.0e-5);
    println!("OK")
}
